lib/reline/line_editor.rb

```diff
@@ -2651,7 +2653,7 @@ def finish
   alias_method :kill_whole_line, :em_kill_line
 
   private def em_delete(key)
-    if (not @is_multiline and @line.empty?) or (@is_multiline and @line.empty? and @buffer_of_lines.size == 1)
+    if @line.empty? and (not @is_multiline or @buffer_of_lines.size == 1) and key == "\C-D"
```

Comment:
`"\C-d".ord`(lowercase d) is better because lowercase is used in this file.
Could you add a test to check that `\C-d` with empty line will end the input?
I think we can use `assert_line(nil)`

Reply:
Thank you for pointing that out, I found a test in `test/reline/test_key_actor_vi.rb` that was testing the same case: `\C-d` with an empty line ends the input, so I added it to `test/reline/test_key_actor_emacs.rb` as well.